Code/Framework/AzToolsFramework/AzToolsFramework/UI/PropertyEditor/EntityPropertyEditor.cpp

```diff
@@ -742,6 +742,19 @@ namespace AzToolsFramework
             ClearComponentEditorSelection();
             ClearComponentEditorState();
         }
+
+        // Clear the focus of the current widget in order to trigger editing
+        // completion before the selected entity list is modified. Otherwise,
+        // the entity will not automatically be marked as dirty since it will
+        // no longer be selected when receiving the BeforePropertyModified event
+        if (DoesOwnFocus())
+        {
+            QWidget* widget = QApplication::focusWidget();
+            if (this != widget)
```

Comment:
By 'last focused widget', I mean the result of QApplication::focusWidget(); If you look at that function, it's what it does - fetch the last focused widget.
So I understood the logic as, 'if the current widget is an ancestor of last focused widget, then clear the focus on the last focused widget'......which indicates that this may not be the best place to clear the focus.

Also , `!AzToolsFramework::ComponentModeFramework::InComponentMode()` indicates that ClearComponentEditorState() is called everytime **except** in component mode

Reply:
Even if !AzToolsFramework::ComponentModeFramework::InComponentMode() is called every time except in component mode, I believe we still want to do the new logic of clearing the focus in all the modes.